File: src/DeepLearning.Domain/Enums/SubmissionStatus.cs
namespace DeepLearning.Domain.Enums
{
    public enum SubmissionStatus
    {
        draft,
        submitted,
        grading,
        grading_failed,
        graded,
        under_dispute,
        standard_revised,
        archived,
        grading_abandoned,
        // Appended last on purpose: the frontend mirrors this enum by ordinal (see
        // enums.ts), and JSON serialises it as a number, so a mid-enum insert would shift
        // archived/grading_abandoned. A score_challenge follow-up thread closed with
        // decision=adjust: the disputed dimension's Band was rewritten in place and a
        // GradingResultRevision audit row written. Distinct from graded so the UI can badge
        // "已改判"; a re-graded submission can still be challenged again (regraded -> under_dispute).
        regraded
    }
}

File: src/DeepLearning.Web/src/lib/submission-status.ts
import { SubmissionStatus } from "@/lib/types/enums";

/**
 * 镜像后端 Domain/Entities/Submission.cs 的 IsResultVisible —— 哪些状态算"已出批改结果"。
 * under_dispute 也算在内:一条追问线程存续期间 submission 会停在 under_dispute(见
 * follow-up-panel.tsx),此时批改结果区域和追问入口都要照常显示。后端/前端各语言各一份定义,
 * 跨语言这条无法避免,但至少前端内部两处(submission-page/grading-result-panel)不再各写一份。
 */
export function isResultVisible(status: number): boolean {
  return (
    status === SubmissionStatus.graded ||
    status === SubmissionStatus.regraded ||
    status === SubmissionStatus.standard_revised ||
    status === SubmissionStatus.under_dispute
  );
}

/** 镜像后端 Submission.IsGradingInProgress —— 哪些状态代表批改正在进行中。 */
export function isGradingInProgress(status: number): boolean {
  return status === SubmissionStatus.submitted || status === SubmissionStatus.grading;
}
